### ISSUE TITLE
Theme Juice: a local WordPress development toolkit

### ISSUE THREAD

**Issue by ezekg** [thechangelog/ping#562]
Hey guys!

Love the podcasts. Been a listener of the changelog for a few years now. You guys have been a huge motivator for me pursuing and creating open source software. I wanted to reach out and let you guys know of a WordPress development toolkit I've built.

It all started a little over 3 years ago when I began using Vagrant for setting up local development environments. I started out utilizing VVV, like most WordPress developers do. At the time, that was whole a lot to take in for a developer who at the time primarily worked with front end tech.

As I started getting more comfortable with the Vagrant ecosystem, I began getting a little annoyed with the constant repetition needed to set up a new local WordPress development project. Thus, a multi-hundred line Rakefile was born! As you can imagine, that didn’t last long.

A few weeks into using that script, which was checked into each repository it was used on (bad!), I started to get annoyed (again) with keeping that script up to date. Over those few weeks of using Rake, I got relatively comfortable with Ruby. So, I started to migrate it into a gem so that I could keep it nice and versioned.

Thus, [Theme Juice, or `tj`](http://cli.themejuice.it), was born. The name came from an old internal project we had that didn't pan out, and I used it as a placeholder. It ended up sticking like all placeholder names do. Theme Juice started out pretty messy, but over the course of 2 years and a few rewrites it's improved a ton with countless features, good test coverage and a solid build system.

Anyways, onto the what! That introduction was a bit longer than I planned, but that’s a short snippet of the history of this project. So, what is Theme Juice? Theme Juice is a local WordPress dev toolkit built on awesome open source tools. With a single command,

``` bash
tj create
```

you can have a new local development site up and running [in a couple minutes](https://asciinema.org/a/85303). It uses Vagrant to create [a robust development environment](https://github.com/ezekg/graft), and lets you create and manage multiple projects within a single virtual machine. On top of that, it can handle deploying your site over SSH using Capistrano, help manage build tools and more.

I didn't intend for this to be such a long post, but I hope it was a nice introduction to the toolkit. I've continued working on all of this in hopes that it could somehow help push WordPress development forward, as it's not exactly an ecosystem that has embraced version control, reproducable deployments and all those good practices other ecosystems utilize.

(As an aside, I just finished a desktop app built on top of the CLI and VM mentioned above, [available here](https://www.themejuice.it), but that’s a closed source product. 👀)

I’d love to chat about the project if you’re interested! I’ve had a tough time gaining a lot of traction, as WordPress doesn’t garner the best from people. Would appreciate any thoughts you have.

Thanks!

**Comment by jerodsanto** [comment]
@ezekg thanks for listening for so long, we love hearing from people we've helped along the way!

I just queued up a [tweet](https://twitter.com/changelog) in your direction and we'll also link this up in the next issue of [Changelog Weekly](https://changelog.com/weekly/).

I don't think it's quite right for the podcast, but if you'd like to write a guest post for changelog.com telling Theme Juice's story and value, I'd be open to that idea. Just holler 💚
